[LST] warning: 'dBeta' may be used uninitialized

While building some unrelated code with the latest IBs using GCC 14 I spotted this warning:

```
In function 'addQuadrupletToMemory',
    inlined from 'operator().isra' at src/RecoTracker/LSTCore/src/alpaka/Quadruplet.h:670:42:
src/RecoTracker/LSTCore/src/alpaka/Quadruplet.h:79:42: warning: 'dBeta' may be used uninitialized [-Wmaybe-uninitialized]
   79 |     quadruplets.dBeta()[quadrupletIndex] = dBeta;
      |                                          ^
src/RecoTracker/LSTCore/src/alpaka/Quadruplet.h: In function 'operator().isra':
src/RecoTracker/LSTCore/src/alpaka/Quadruplet.h:614:35: note: 'dBeta' was declared here
  614 |               float rzChiSquared, dBeta, nonAnchorChiSquared, regressionCenterX, regressionCenterY, regressionRadius,
      |                                   ^
In function 'addQuadrupletToMemory',
    inlined from 'operator().isra' at src/RecoTracker/LSTCore/src/alpaka/Quadruplet.h:792:38:
src/RecoTracker/LSTCore/src/alpaka/Quadruplet.h:79:42: warning: 'dBeta' may be used uninitialized [-Wmaybe-uninitialized]
   79 |     quadruplets.dBeta()[quadrupletIndex] = dBeta;
      |                                          ^
src/RecoTracker/LSTCore/src/alpaka/Quadruplet.h: In function 'operator().isra':
src/RecoTracker/LSTCore/src/alpaka/Quadruplet.h:739:31: note: 'dBeta' was declared here
  739 |           float rzChiSquared, dBeta, nonAnchorChiSquared, regressionCenterX, regressionCenterY, regressionRadius,
      |                               ^
```

From a quick look at `RecoTracker/LSTCore/src/alpaka/Quadruplet.h` it seems legit, because `dBeta` is not always computed, and may be left uninitialised.

It should be enough to initialise it to some known invalid value, or even just `0` if the value is not used later in the code.

--- a/RecoTracker/LSTCore/src/alpaka/Quadruplet.h
+++ b/RecoTracker/LSTCore/src/alpaka/Quadruplet.h
@@ -464,6 +464,8 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::lst {
                                              dBeta,
                                              ptCut))
         return false;
+    } else {
+      dBeta = 0;
     }
 
     if (not passT4RZConstraint(acc,

